src/Standards/Generic/Sniffs/Metrics/CyclomaticComplexitySniff.php

```diff
@@ -60,7 +60,7 @@ public function process(File $phpcsFile, $stackPtr)
     {
         $tokens = $phpcsFile->getTokens();
 
-        // Ignore abstract methods.
+        // Ignore abstract and interface methods. Bail early when live coding.
         if (isset($tokens[$stackPtr]['scope_opener']) === false) {
```

Comment:
My understanding is that it only makes sense to run this sniff when the function has a scope opener and a scope closer so that it knows when to start counting cyclomatic complexity and when it should stop.

That being said, as far as I can test, `scope_opener` and `scope_closer` are only set when both are present in the code. So it seems to me that it is fine to check just for `scope_opener` as it is done in this sniff.

Do you agree? I have a vague memory of us discussing something similar in the past, but I don't recall the details, and I could not find it.

I checked and saw an example of a sniff that use `scope_closer`, but only check if `scope_opener` is set and an example of a sniff that check for both:

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/9bf4e15b9ddfc16d162670702f54dc09fb0345a1/src/Standards/Generic/Sniffs/PHP/LowerCaseTypeSniff.php#L94-L98

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/84acf4e56f110db8e75cb9a575c5727df637643c/src/Standards/Generic/Sniffs/Metrics/NestingLevelSniff.php#L60-L66